src/ocm/components/hosts/ClusterHostsTable.tsx

```diff
@@ -36,7 +36,7 @@ const ClusterHostsTable: React.FC<ClusterHostsTableProps> = ({
 
   const content = React.useMemo(
     () => [
-      hostnameColumn(onEditHost),
+      hostnameColumn(onEditHost, undefined, !actionChecks.canEditHostname()),
```

Comment:
to stay consistent with other columns, do not evaluate `canEditHostname` here, just pass the callback

```suggestion
      hostnameColumn(onEditHost, undefined, actionChecks.canEditHostname),
```